UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -407,203 +400,115 @@ protected virtual MaterialCacheData CreateMaterial(GLTF.Schema.Material def, int
-					material.SetInt("_DstBlend", (int) UnityEngine.Rendering.BlendMode.OneMinusSrcAlpha);
-					material.SetInt("_ZWrite", 0);
-					material.DisableKeyword("_ALPHATEST_ON");
-					material.EnableKeyword("_ALPHABLEND_ON");
-					material.DisableKeyword("_ALPHAPREMULTIPLY_ON");
-					material.renderQueue = (int) UnityEngine.Rendering.RenderQueue.Transparent;
-				}
+				IUniformMap mapper;
+				const string specGlossExtName = KHR_materials_pbrSpecularGlossinessExtensionFactory.EXTENSION_NAME;
+				if (_root.ExtensionsUsed != null && _root.ExtensionsUsed.Contains(specGlossExtName)
+					&& def.Extensions != null && def.Extensions.ContainsKey(specGlossExtName))
+					mapper = new SpecGlossMap(MaximumLod);
 				else
-				{
-					material.SetOverrideTag("RenderType", "Opaque");
-					material.SetInt("_SrcBlend", (int) UnityEngine.Rendering.BlendMode.One);
-					material.SetInt("_DstBlend", (int) UnityEngine.Rendering.BlendMode.Zero);
-					material.SetInt("_ZWrite", 1);
-					material.DisableKeyword("_ALPHATEST_ON");
-					material.DisableKeyword("_ALPHABLEND_ON");
-					material.DisableKeyword("_ALPHAPREMULTIPLY_ON");
-					material.renderQueue = -1;
-				}
+					mapper = new MetalRoughMap(MaximumLod);
 
-				if (def.DoubleSided)
-				{
-					material.SetInt("_Cull", (int) CullMode.Off);
-				}
-				else
-				{
-					material.SetInt("_Cull", (int) CullMode.Back);
-				}
+				mapper.AlphaMode = def.AlphaMode;
+				mapper.DoubleSided = def.DoubleSided;
 
-				if (def.PbrMetallicRoughness != null)
+				var mrMapper = mapper as IMetalRoughUniformMap;
+				if (def.PbrMetallicRoughness != null && mrMapper != null)
 				{
 					var pbr = def.PbrMetallicRoughness;
 
-					material.SetColor("_Color", pbr.BaseColorFactor.ToUnityColor());
+					mrMapper.BaseColorFactor = pbr.BaseColorFactor.ToUnityColor();
 
 					if (pbr.BaseColorTexture != null)
 					{
 						var textureDef = pbr.BaseColorTexture.Index.Value;
-						material.SetTexture("_MainTex", CreateTexture(textureDef));
+						mrMapper.BaseColorTexture = CreateTexture(textureDef);
+						mrMapper.BaseColorTexCoord = pbr.BaseColorTexture.TexCoord;
 
-						ApplyTextureTransform(pbr.BaseColorTexture, material, "_MainTex");
+						//ApplyTextureTransform(pbr.BaseColorTexture, material, "_MainTex");
 					}
 
-					material.SetFloat("_Metallic", (float) pbr.MetallicFactor);
+					mrMapper.MetallicFactor = pbr.MetallicFactor;
 
 					if (pbr.MetallicRoughnessTexture != null)
 					{
 						var texture = pbr.MetallicRoughnessTexture.Index.Value;
-						material.SetTexture("_MetallicRoughnessMap", CreateTexture(texture));
+						mrMapper.MetallicRoughnessTexture = CreateTexture(texture);
+						mrMapper.MetallicRoughnessTexCoord = pbr.MetallicRoughnessTexture.TexCoord;
 
-						ApplyTextureTransform(pbr.MetallicRoughnessTexture, material, "_MetallicRoughnessMap");
+						//ApplyTextureTransform(pbr.MetallicRoughnessTexture, material, "_MetallicRoughnessMap");
```

Comment:
?

Reply:
I tentatively removed support for that extension until we can figure out where we should put it. I'm invested in getting that right, its disabling is only temporary.